src/Elements/Input.react.js

```diff
@@ -207,15 +207,21 @@ class Input extends Component {
                         newValue = Math.max(min, newValue);
                     }
                 }
-            }
+
+                this.setState({ value: newValue });
```

Comment:
`this.setState({ value: newValue }); needs to be in an else if `onChange` is undefined.

Reply:
Just to confirm ... it _doesn't_ need to set its own internal state if the `onChange` prop is specified?  I guess we're relying on the consuming component to pass the new value in via props (i.e. the consuming component sets some of its own state or whatever is bound to the `value` prop of the `<Input>` in its function that handles `onChange`)?

Reply:
Yes, exactly.

Reply:
Fixed